pkg/component/etcd/configmap/configmap.go

```diff
@@ -0,0 +1,208 @@
+    # The URLs needed to be a comma-separated list.
+    advertise-client-urls: ` + protocol + `://0.0.0.0:` + clientPort + `
+
+    # List of comma separated URLs to listen on for peer traffic.
+    listen-peer-urls: ` + protocol + `://0.0.0.0:` + serverPort + `
+
+    # List of this member's peer URLs to advertise to the public.
+    # The URLs needed to be a comma-separated list.
+    initial-advertise-peer-urls: ` + c.values.PeerServiceName + `@` + c.values.EtcdNameSpace + `@` + serverPort + `
+
+    # Initial cluster token for the etcd cluster during bootstrap.
+    initial-cluster-token: initial
+
+    # Initial cluster state ('new' or 'existing').
+    initial-cluster-state: new
+
+    # Initial cluster
+    initial-cluster: ` + c.values.InitialCluster + `
+
+    # auto-compaction-mode ("periodic" or "revision").`
+
+		autoCompactionMode := "periodic"
+		if c.values.AutoCompactionMode != nil {
+			autoCompactionMode = string(*c.values.AutoCompactionMode)
+		}
+		etcdConfigYaml = etcdConfigYaml + `
+    auto-compaction-mode: ` + autoCompactionMode
+
+		autoCompactionRetention := "30m"
+		if c.values.AutoCompactionRetention != nil {
+			autoCompactionRetention = string(*c.values.AutoCompactionRetention)
+		}
+		etcdConfigYaml = etcdConfigYaml + `
+    # auto-compaction-retention defines Auto compaction retention length for etcd.
+    auto-compaction-retention: ` + autoCompactionRetention
+
+		if enableTLS {
+			etcdConfigYaml = etcdConfigYaml + `
+    client-transport-security:
+      # Path to the client server TLS cert file.
+      cert-file: /var/etcd/ssl/server/tls.crt
+
+      # Path to the client server TLS key file.
+      key-file: /var/etcd/ssl/server/tls.key
+
+      # Enable client cert authentication.
+      client-cert-auth: true
+
+      # Path to the client server TLS trusted CA cert file.
+      trusted-ca-file: /var/etcd/ssl/ca/ca.crt
+
+      # Client TLS using generated certificates
+      auto-tls: false`
+		}
+		cm.ObjectMeta = getObjectMeta(c.values)
+		cm.OwnerReferences = getOwnerReferences(c.values)
+		cm.Data = map[string]string{"etcd.conf.yaml": etcdConfigYaml}
+		return nil
+	})
+	return err
+
+}
+
+func (c *component) Destroy(ctx context.Context) error {
+	configMap := c.emptyConfigmap(c.values.ConfigMapName)
+
+	if err := c.deleteConfigmap(ctx, configMap); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (c *component) deleteConfigmap(ctx context.Context, cm *corev1.ConfigMap) error {
+	return client.IgnoreNotFound(c.client.Delete(ctx, cm))
+}
+
+func (c *component) emptyConfigmap(name string) *corev1.ConfigMap {
+	return &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: c.namespace,
+		},
+	}
+}
+
+func getObjectMeta(val Values) metav1.ObjectMeta {
+	labels := map[string]string{"name": "etcd", "instance": val.EtcdName}
+	return metav1.ObjectMeta{
+		Name:      fmt.Sprintf("etcd-bootstrap-%s", string(val.EtcdUID[:6])),
```

Comment:
Can you extract this to a function as done for the other `Get*Name` functions?

Reply:
Done